### ISSUE TITLE
stepfunctions-api: resulting api will not pass path

### ISSUE THREAD

**Issue by thomasfinneyjr** [aws/aws-cdk#21916]
### Describe the bug

The construct fails to create the api with a proxy resource. As a result only the root path is usable, any child paths fail to function as described in the documentation: 
[https://github.com/aws/aws-cdk/blob/56ba2ab2c2d9240b76ece17c3296488a63f0b232/packages/%40aws-cdk/aws-apigateway/README.md?plain=1#L182-L200](url)

### Expected Behavior

As per the documentation:  
[https://github.com/aws/aws-cdk/blob/56ba2ab2c2d9240b76ece17c3296488a63f0b232/packages/%40aws-cdk/aws-apigateway/README.md?plain=1#L182-L200](url)

child paths should be passed to the API gateway.

### Current Behavior

Currently attempting to pass ant path other than the root path results in an authorization error (which really is an incorrectly handled exception) the error is thrown because there is no resource in the created API which can resolve the path.

### Reproduction Steps

1. Create the api using the exact code provided in the documentation: [https://github.com/aws/aws-cdk/blob/56ba2ab2c2d9240b76ece17c3296488a63f0b232/packages/%40aws-cdk/aws-apigateway/README.md?plain=1#L148-L162](url)
2. Attempt to call the api at any child path. Such as the example in the documentation here: [ https://github.com/aws/aws-cdk/blob/56ba2ab2c2d9240b76ece17c3296488a63f0b232/packages/%40aws-cdk/aws-apigateway/README.md?plain=1#L182-L200](url)

### Possible Solution

The construct should also create a greedy proxy which behaves the same as the root resource. 

### CDK CLI Version

2.40.0

### Node.js Version

16

### OS

MacOS

### Language

Typescript

**Comment by peterwoodworth** [comment]
You're right I was able to reproduce this, thanks for reporting this @thomasfinneyjr 

Your proposed solution should fix this 🙂
